layers/state_tracker/cmd_buffer_state.cpp

```diff
@@ -1078,7 +1070,21 @@ void CMD_BUFFER_STATE::UpdatePipelineState(CMD_TYPE cmd_type, const VkPipelineBi
     const auto lv_bind_point = ConvertToLvlBindPoint(bind_point);
     auto &last_bound = lastBound[lv_bind_point];
     PIPELINE_STATE *pipe = last_bound.pipeline_state;
-    if (pipe && VK_NULL_HANDLE != last_bound.pipeline_layout) {
+    if (!pipe) {
+        return;
+    }
+    if (pipe->vertex_input_state && !pipe->vertex_input_state->binding_descriptions.empty()) {
+        vertex_buffer_used = true;
+    }
+
+    // Update the consumed viewport/scissor count.
+    {
+        usedViewportScissorCount = std::max({usedViewportScissorCount, pipelineStaticViewportCount, pipelineStaticScissorCount});
+        usedDynamicViewportCount |= pipe->IsDynamic(VK_DYNAMIC_STATE_VIEWPORT_WITH_COUNT);
+        usedDynamicScissorCount |= pipe->IsDynamic(VK_DYNAMIC_STATE_SCISSOR_WITH_COUNT);
+    }
+
+    if (last_bound.pipeline_layout != VK_NULL_HANDLE) {
```

Comment:
You can return early. This avoids double nested logic.

```suggestion
    if (last_bound.pipeline_layout == VK_NULL_HANDLE) {
        return;
    }
```

Reply:
so the issue is when we decide to add more logic under to add to `UpdatePipelineState` we now have to be careful, either make sure to add it above, or risk a bug someone didn't see this sudden `return` case in the middle of the function (I have hit this a few times before in the VVL with early returns mid-functions)

I think if we are nesting to much, it should be pulled out to another function

Reply:
> or risk a bug someone didn't see this sudden return case in the middle of the function

I see your point. But also that is exactly why we try to require tests when writing validation.